guides/common/modules/con_puma-workers-and-threads-autotuning.adoc

```diff
@@ -0,0 +1,35 @@
+[id="Puma_Workers_and_Threads_Auto_Tuning_{context}"]
+= Puma Workers and Threads Auto-Tuning
+
+If you do not provide any Puma workers and thread values with `{foreman-installer} or they are not present in your {Project} configuration, the `{foreman-installer}` configures a balanced number of workers.
+It follows this formula:
+
+----
+min(CPU*1.5, RAM_IN_GB - 1.5)
+----
+
+// what does this mean?
+// OOM = out of memory?
+// what is 'wrt'?
+which is too much wrt. memory - there have been cases where too many workers triggered OOM on {Project}.
+
+This should be fine for most cases, but with some usage patterns tuning is needed to either limit the amount of resources dedicated to Puma (so other {Project} components can use these) or for any other reason.
+Each Puma worker consumes around 1 GiB of RAM.
+
+// What are those two commands necessary for?
+// See the current config and Process ID?
+// ps -p _My_Puma_Process_ID_ -o command
+// Why do you want to see this?
+For your current setting see this:
+
+----
+# cat /etc/systemd/system/foreman.service.d/installer.conf
+[Service]
+User=foreman
+Environment=FOREMAN_ENV=production
+Environment=FOREMAN_HOME=/usr/share/foreman
+Environment=FOREMAN_PUMA_THREADS_MIN=5
+Environment=FOREMAN_PUMA_THREADS_MAX=5
+Environment=FOREMAN_PUMA_WORKERS=30
+# pgrep -u foreman --list-full | grep 'puma: cluster worker'
```

Comment:
Wouldn't it be easier to see the progress tree using `systemctl status foreman.service`?

Reply:
Yes, added.